stubs/icalendar/icalendar/cal.pyi

```diff
@@ -0,0 +1,82 @@
+from _typeshed import Incomplete, SupportsItems
+from typing import Any, ClassVar, Final, Literal, overload
+
+from pytz.tzinfo import DstTzInfo
+
+from .caselessdict import CaselessDict
+from .parser import Contentline, Contentlines
+from .prop import TypesFactory
+
+class ComponentFactory(CaselessDict[Incomplete]):
+    def __init__(self, *args, **kwargs) -> None: ...
+
+INLINE: CaselessDict[int]
+
+class Component(CaselessDict[Incomplete]):
+    name: ClassVar[str | None]
+    required: ClassVar[tuple[str, ...]]
+    singletons: ClassVar[tuple[str, ...]]
+    multiple: ClassVar[tuple[str, ...]]
+    exclusive: ClassVar[tuple[str, ...]]
+    inclusive: ClassVar[tuple[tuple[str, ...], ...]]
+    ignore_exceptions: ClassVar[bool]
+    subcomponents: list[Incomplete]
+    errors: list[str]
+
+    def __init__(self, *args, **kwargs) -> None: ...
+    def __bool__(self) -> bool: ...
+    __nonzero__ = __bool__
+    def is_empty(self) -> bool: ...
+    @property
+    def is_broken(self) -> bool: ...
+    @overload
+    def add(self, name: str, value: Any, *, encode: Literal[False]) -> None: ...
```

Comment:
Aren't these overloads redundant? Since the last overload has `encode: bool`, I think it encompasses all possible calls on the other two overloads.